website/blog/2024/04-23-scrapy-vs-crawlee/index.md

```diff
@@ -0,0 +1,323 @@
+
+Scrapy does not have autoscaling capabilities inbuilt, but it can be done using external services like [Scrapyd](https://scrapyd.readthedocs.io/en/latest/) or deployed in a distributed manner with Scrapy Cluster.
+
+Crawlee has [built-in autoscaling](https://crawlee.dev/api/core/class/AutoscaledPool) with `AutoscaledPool`. It increases the number of requests that are processed concurrently within one crawler.
+
+### Queue Management
+
+Scrapy supports both breadth-first and depth-first crawling strategies using a disk-based queuing system. By default, it uses the LIFO queue for the pending requests, which means it is using depth-first order, but if you want to use breadth-first order, you can do it by changing these settings:
+
+```py
+DEPTH_PRIORITY = 1 
+SCHEDULER_DISK_QUEUE = "scrapy.squeues.PickleFifoDiskQueue" 
+SCHEDULER_MEMORY_QUEUE = "scrapy.squeues.FifoMemoryQueue"
+```
+
+Crawlee uses breadth-first by default and you can override it on a per-request basis by using the `forefront: true` argument in `addRequest` and its derivatives. If you use `forefront: true` for all requests, it becomes a depth-first process.
+
+### CLI Support
+
+Scrapy has a [powerful command-line interface](https://docs.scrapy.org/en/latest/topics/commands.html#command-line-tool) that offers functionalities like starting a project, generating spiders, and controlling the crawling process.
+
+Scrapy CLI comes with Scrapy installation. Just run this command, and you are good to go:
+
+`pip install scrapy`
+
+Crawlee also [includes a CLI tool](https://crawlee.dev/docs/quick-start#installation-with-crawlee-cli) (`crawlee-cli`) that facilitates project setup, crawler creation and execution, streamlining the development process for users familiar with Node.js environments. The command for installation is: 
+
+`npx crawlee create my-crawler`
+ 
+### Proxy Rotation and Storage Management
+
+Scrapy handles it via custom middleware. You have to install their [`scrapy-rotating-proxies`]((https://pypi.org/project/scrapy-rotating-proxies/)) package using pip. 
+
+`pip install scrapy-rotating-proxies`
+
+Then in the `settings.py` file add `ROTATING_PROXY_LIST` and the middleware to the `DOWNLOADER_MIDDLEWARES` and specify the list of proxy servers. For example:
+
+```py
+DOWNLOADER_MIDDLEWARES = {
+    # Lower value means higher priority
+    'scrapy.downloadermiddlewares.retry.RetryMiddleware': 90,
+    'scrapy_rotating_proxies.middlewares.RotatingProxyMiddleware': 610,
+    'scrapy_rotating_proxies.middlewares.BanDetectionMiddleware': 620,
+}
+
+ROTATING_PROXY_LIST = [
+    'proxy1.com:8000',
+    'proxy2.com:8031',
+    # Add more proxies as needed
+]
+```
+Now create a spider with the code you want to scrape any site and the `ROTATING_PROXY_LIST` in `settings.py` will manage which proxy to use for each request.
```

Comment:
Would be great to mention how the proxy rotation work. Is it round robin? Is there some logic behind it?